core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    firstCollectionNode = CollectionNodeObjectFactory.create(
+        firstCollectionNodeBackendObject);
+
+    var secondCollectionNodeBackendObject = {
+      exploration_id: 'exp_id1',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title1',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    secondCollectionNode = CollectionNodeObjectFactory.create(
+        secondCollectionNodeBackendObject);
+
+    var thirdCollectionNodeBackendObject = {
+      exploration_id: 'exp_id2',
+      prerequisite_skills: [],
+      acquired_skills: [],
+      exploration_summary: {
+        title: 'exp title2',
+        category: 'exp category',
+        objective: 'exp objective'
+      }
+    };
+    thirdCollectionNode = CollectionNodeObjectFactory.create(
+        thirdCollectionNodeBackendObject);
+  }));
+
+  var createLinearCollection = function() {
+    var collection = CollectionObjectFactory.createEmptyCollection();
+    firstCollectionNode.getAcquiredSkillList().setSkills(['exp title0']);
+    secondCollectionNode.getPrerequisiteSkillList().setSkills(['exp title0']);
+    secondCollectionNode.getAcquiredSkillList().setSkills(['exp title1']);
+    thirdCollectionNode.getPrerequisiteSkillList().setSkills(['exp title1']);
+    thirdCollectionNode.getAcquiredSkillList().setSkills(['exp title2']);
+
+    // Randomize the add order for robust testing since linearity depends on
+    // skills and not the order in which the nodes where added.
+    collection.addCollectionNode(thirdCollectionNode);
+    collection.addCollectionNode(secondCollectionNode);
+    collection.addCollectionNode(firstCollectionNode);
+    return collection;
+  };
+
+  var createNonLinearCollection = function() {
+    var collection = CollectionObjectFactory.createEmptyCollection();
+    firstCollectionNode.getAcquiredSkillList().setSkills(['exp title0']);
+    secondCollectionNode.getPrerequisiteSkillList().setSkills(['exp title0']);
+    secondCollectionNode.getAcquiredSkillList().setSkills(['exp title1']);
+    thirdCollectionNode.getPrerequisiteSkillList().setSkills(['exp title0']);
+    thirdCollectionNode.getAcquiredSkillList().setSkills(['exp title2']);
+    collection.addCollectionNode(firstCollectionNode);
+    collection.addCollectionNode(secondCollectionNode);
+    collection.addCollectionNode(thirdCollectionNode);
+    return collection;
+  };
+
+  describe('removeCollectionNode()', function() {
+    it('should not remove a non-existent node from a single node collection',
+        function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      collection.addCollectionNode(firstCollectionNode);
+      expect(collection.containsCollectionNode('exp_id0')).toBe(true);
+      expect(CollectionLinearizerService.removeCollectionNode(collection,
```

Comment:
For consistency, please break immediately after ``(`` if you run out of room. Please fix here and elsewhere.